README.md

```diff
@@ -66,55 +66,65 @@ flowchart LR
     Handle -->|Decode Init Errors| Decoder
 
     Context -->|Run| Library
-    Context -->|Store Go References| ContextAllocator
+    Context -->|Store Go References| CGORefPool
 
-    Encoder -->|Allocate Waf Objects| EncoderAllocator
+    Encoder -->|Allocate Waf Objects| TempCGORefPool
 
-    EncoderAllocator -->|Copy after each encoding| ContextAllocator
+    TempCGORefPool -->|Copy after each encoding| CGORefPool
 
     Library -->|Call C code| libddwaf
 
     classDef hidden display: none;
 ```
 
-### Allocator
+### CGO Reference Pool
 
-The cgoRefPool is a pure Go cgoRefPool of `ddwaf_object` C values on the Go memory heap.
-the `cgoRefPool` go type is a way to make sure we can safely send go allocated data to the C side of the WAF
+The cgoRefPool type is a pure Go pointer pool of `ddwaf_object` C values on the Go memory heap.
+the `cgoRefPool` go type is a way to make sure we can safely send Go allocated data to the C side of the WAF
 The main issue is the following: the `wafObject` uses a C union to store the tree structure of the full object,
 union equivalent in go are interfaces and they are not compatible with C unions. The only way to be 100% sure
 that the Go `wafObject` struct has the same layout as the C one is to only use primitive types. So the only way to
 store a raw pointer is to use the `uintptr` type. But since `uintptr` do not have pointer semantics (and are just
-basically integers), we need another structure to store the value as Go pointer because the GC is lurking. That's
-where the `cgoRefPool` object comes into play: all new `wafObject` elements are created via this API whose especially
+basically integers), we need another method to store the value as Go pointer because the GC will delete our data if it
+not reference bu Go pointers.
```

Comment:
```suggestion
basically integers), we need another method to store the value as Go pointer because the GC will delete our data if it
is not referenced by Go pointers.
```